packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -1505,9 +1505,10 @@ function build(
   }
 
   /**
+   * @param { import('../types-external.js').DisconnectObject } disconnectObject
    * @returns {Promise<void>}
    */
-  async function stopVat() {
+  async function stopVat(disconnectObject) {
```

Comment:
We need to give `disconnectObject` to `watchedPromiseManager.prepareShutdownRejections()` too, and change the implementation to store the serialized DO in the DB next to `deadPromises`, and change `loadWatchedPromiseTable` to use it for the rejections instead of the current hardcoded `VatUpgradedMessage = 'vat upgraded'`, so that all promises (locally-decided locally-watched ones and locally-decided remotely-followed ones) are all rejected in the same way.

The stored DO should be deleted during `loadWatchedPromiseTable` at the same time it deletes `deadPromises`.

That also means we need to update `test-promise-watcher.js` to look for the more detailed DisconnectionObject.

Reply:
Oy.  Will do.